studio/src/lib/schema-helpers.ts

```diff
@@ -156,6 +157,75 @@ export const mapGraphQLType = (
   throw new Error("Unsupported GraphQL type");
 };
 
+export const extractVariablesFromGraphQL = (
+  body: string,
+  ast: GraphQLSchema | null,
+) => {
+  const allTypes = ast
+    ? Object.values(ast.getTypeMap())
+        .filter((type) => !type.name.startsWith("__"))
+        .sort()
+    : [];
+
+  const variablesRegex =
+    /\$([a-zA-Z_][a-zA-Z0-9_]*)\s*:\s*([^\s!]+)(!)?(\s*=\s*([^,\)]+))?/g;
+  let variables: Record<string, any> = {};
+
+  let match;
+  while ((match = variablesRegex.exec(body)) !== null) {
+    const [, variableName, variableType, nonNull, , defaultValue] = match;
+    let defaultValueParsed;
+    if (defaultValue !== undefined && defaultValue !== "") {
+      defaultValueParsed = JSON.parse(defaultValue);
+    } else {
+      defaultValueParsed = nonNull
+        ? getDefaultValue(variableType, allTypes)
+        : null;
+    }
+
+    variables[variableName] = defaultValueParsed;
+  }
+
+  return variables;
+};
+
+function getDefaultValue(
```

Comment:
Please add a few tests for each of the methods.

Reply:
Added. There is no ui change. variables are just added to the snippet you copy